pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -243,16 +268,20 @@ func (r *Reconciler) exists() (bool, error) {
 	if err := r.validateZone(); err != nil {
 		return false, fmt.Errorf("unable to verify project/zone exists: %v/%v; err: %v", r.projectID, zone, err)
 	}
-	_, err := r.computeService.InstancesGet(r.projectID, zone, r.machine.Name)
-	if err == nil {
-		klog.Infof("Machine %q already exists", r.machine.Name)
-		return true, nil
+
+	instance, err := r.computeService.InstancesGet(r.projectID, r.providerSpec.Zone, r.machine.Name)
+	if err != nil {
+		e, ok := err.(*googleapi.Error)
+		if ok && e.Code == 404 {
```

Comment:
We have a function for this.

Reply:
I inlined it as it was only called once.